### PR TITLE
add support for external generators

### PR DESCRIPTION
**Pull request by eu-ch**
Hey @dromer, this is the external generator feature we've discussed earlier.

### CODE REVIEW

**Comment by dromer** [comment]
Cool! from first look it seems alright.

I'm not 100% yet on calling the abstract class `Compiler`. Since atm this specific class is only used for Generators.
So maybe it should be called `Generator` instead :#

`CompilerResp` is used in various places where objects have a `compile()` method though. So I think I named it this because of that.

What do you think, should we name this specific group of classes `Generator`? I think it would be more explicit.

File: hvcc/__init__.py [hunk 1]
```diff
@@ -373,6 +388,14 @@ def compile_dataflow(
             print("--> Generating Wwise plugin")
         results["c2wwise"] = c2wwise.c2wwise.compile(**gen_args)
 
+    if ext_generators:
+        for module_name in ext_generators:
+            compiler = load_ext_generator(module_name, verbose)
+            if compiler:
```
Comment (dromer) [inline hvcc/__init__.py:394]: This should say `if compiler is not None:`

**Comment by dromer** [comment]
btw I'm also going to move the Meta types to the new `hvcc.types` section :)

**Comment by dromer** [comment]
ok, I made these changes, please adjust the code so it fits again :)

File: hvcc/__init__.py [hunk 2]
```diff
@@ -409,6 +432,11 @@ def main() -> bool:
         nargs="+",
         default=["c"],
         help="List of generator outputs: c, unity, wwise, js, pdext, daisy, dpf, fabric, owl")
+    parser.add_argument(
+        "-G",
```
Comment (dromer) [inline hvcc/__init__.py:436]: Not entirely convinced about using `G` for this as it might be a bit confusing.

**Comment by eu-ch** [comment]
> ok, I made these changes, please adjust the code so it fits again :)

Thanks, will do later today. Also, please make up your mind regarding ‘-G’ argument, so I can change in code and docs. Alternatively, I can omit the short arg in favour of a long ‘--ext-gen’.

**Pull request merged**